server-common/src/main/java/org/apache/kafka/server/common/TestFeatureVersion.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+import java.util.Collections;
+import java.util.Map;
+
+public enum TestFeatureVersion implements FeatureVersion {
+
+    TEST_0(0, MetadataVersion.IBP_3_3_IV0, Collections.emptyMap()),
```

Comment:
Version 0 == "feature doesn't exist", does it need to be explicitly codified the enum?

Reply:
I tried this, but the logic for defaults breaks when we don't have a 0 version. 
```
    val allFeaturesAndLevels: List[FeatureVersion] = allFeatures.map { feature =>
      val level: java.lang.Short = specifiedFeatures.getOrElse(feature.featureName, feature.defaultValue(metadataVersionForDefault))
      feature.fromFeatureLevel(level)
    }
```
Here, I suppose we could do some work to filter out 0s and assume on the rest of the code that a missing value in the map = 0. If that is preferred I can do that instead, but I also don't think the 0 version necessarily hurts anything.

Reply:
I've made this change